--- a/lib/fcntl.in.h
+++ b/lib/fcntl.in.h
@@ -237,7 +237,9 @@
 _GL_CXXALIAS_SYS (openat, int,
                   (int fd, char const *file, int flags, /* mode_t mode */ ...));
 # endif
-_GL_CXXALIASWARN (openat);
+_GL_CXXALIASWARN1 (openat, int,
+                    (int fd, char const *file, int flags,
+                     /* mode_t mode */ ...));
 #elif defined GNULIB_POSIXCHECK
 # if HAVE_RAW_DECL_OPENAT
 _GL_WARN_ON_USE (openat, "openat is not portable - "
