--- a/pcre2-sys/build.rs
+++ b/pcre2-sys/build.rs
@@ -82,6 +82,9 @@
         builder.debug(true);
     }
     builder.compile("libpcre2.a");
+    if target == "aarch64-unknown-linux-musl" {
+        println!("cargo:rustc-link-lib=gcc");
+    }
 }

 fn pcre2_sys_static() -> Option<bool> {
@@ -128,9 +131,6 @@
     if target == "armv7-linux-androideabi" {
         return;
     }
-    if target == "aarch64-unknown-linux-musl" {
-        return;
-    }
     if target.contains("musleabi") {
         return;
     }
